### ISSUE TITLE
"sometimes" rule

### ISSUE THREAD

**Issue by nikinicky** [mikeerickson/validatorjs#69]
In some situations, I may wish to run validation checks against a field only if that field is present in the input array. We know that Laravel validation framework has this feature. Does validatorjs have this feature ?

**Comment by iamdtang** [comment]
No, validatorjs does not have this rule. Want to add it? =)

**Comment by globalchubby** [comment]
Hi, nice library! 

Is someone working on this issue?

I'm not familiar with Laravel; could @nickiesiva or @skaterdav85 post a code snippet that illustrates the desired behavior?

**Comment by iamdtang** [comment]
> In some situations, you may wish to run validation checks against a field only if that field is present in the input array. To quickly accomplish this, add the sometimes rule to your rule list:

https://laravel.com/docs/5.2/validation#conditionally-adding-rules

@yangmillstheory If you'd like to work on it, go for it :)

**Comment by globalchubby** [comment]
OK, I've cloned the repo, built and ran the tests.

What do you think about a section in the REAMDE for guidelines for contributing? I also noticed that you hadn't listed gulp-cli as a `devDependency` but I don't have it (and don't want it) installed globally. We could lock down a version that works specifically with your project.

**Comment by iamdtang** [comment]
There is another file for contributing, but it could be improved. https://github.com/skaterdav85/validatorjs/blob/master/CONTRIBUTE.md

Do you mean grunt instead of gulp? If you have a local version of grunt, is there a way to run it that way? If so, sure you can change that. The scripts section in the package.json will need to be updated.

**Comment by globalchubby** [comment]
Yes, I meant Grunt, thanks. It would install to `node_modules/.bin/grunt`, and developers would be running that specific version instead of what's in their global `node_modules` folder, somewhere in `/usr/local/` in a *nix environment.

If they don't have it installed globally (like me), they get something like

``` shell
validatorjs (master u= ?!) $ grunt
-bash: grunt: command not found
```

I've been using [this shorthand](http://stackoverflow.com/a/15157360/2419669) since I like not prefixing commands with `node_modules/.bin` every time I run a project-specific binary.

**Comment by iamdtang** [comment]
cool, ya I like that better too. If you want to make that update, go for it. :)

**Comment by Patrik-Lundqvist** [comment]
Useful feature, nice work. I hope this will be included in a new release soon!

**Comment by iamdtang** [comment]
@Patrik-Lundqvist Just released the latest as 3.7.0
